src/inspector_coverage.cc

```diff
@@ -0,0 +1,172 @@
+#include "base_object-inl.h"
+#include "debug_utils.h"
+#include "inspector_agent.h"
+#include "node_internals.h"
+#include "v8-inspector.h"
+
+namespace node {
+namespace coverage {
+
+using v8::Context;
+using v8::Function;
+using v8::FunctionCallbackInfo;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::Local;
+using v8::MaybeLocal;
+using v8::NewStringType;
+using v8::Object;
+using v8::ObjectTemplate;
+using v8::String;
+using v8::Value;
+
+using v8_inspector::StringBuffer;
+using v8_inspector::StringView;
+
+std::unique_ptr<StringBuffer> ToProtocolString(Isolate* isolate,
+                                               Local<Value> value) {
+  TwoByteValue buffer(isolate, value);
+  return StringBuffer::create(StringView(*buffer, buffer.length()));
+}
+
+class V8CoverageConnection : public BaseObject {
+ public:
+  class V8CoverageSessionDelegate : public inspector::InspectorSessionDelegate {
+   public:
+    explicit V8CoverageSessionDelegate(V8CoverageConnection* connection)
+        : connection_(connection) {}
+
+    void SendMessageToFrontend(
+        const v8_inspector::StringView& message) override {
+      Environment* env = connection_->env_;
+      Debug(env,
+            DebugCategory::COVERAGE,
+            "Sending message to frontend, ending = %s\n",
+            connection_->ending_ ? "true" : "false");
+      if (!connection_->ending_) {
+        return;
+      }
+      Isolate* isolate = env->isolate();
+      Local<Context> context = env->context();
+
+      HandleScope handle_scope(isolate);
+      Context::Scope context_scope(env->context());
+      MaybeLocal<String> v8string =
+          String::NewFromTwoByte(isolate,
+                                 message.characters16(),
+                                 NewStringType::kNormal,
+                                 message.length());
+      Local<Value> result = v8string.ToLocalChecked().As<Value>();
+      Local<Function> fn = connection_->env_->on_coverage_message_function();
+      CHECK(!fn.IsEmpty());
+      USE(fn->Call(context, v8::Null(isolate), 1, &result));
+    }
+
+   private:
+    V8CoverageConnection* connection_;
+  };
+
+  SET_MEMORY_INFO_NAME(V8CoverageConnection)
+  SET_SELF_SIZE(V8CoverageConnection)
+
+  void MemoryInfo(MemoryTracker* tracker) const override {
+    tracker->TrackFieldWithSize(
+        "session", sizeof(*session_), "InspectorSession");
+  }
+
+  explicit V8CoverageConnection(Environment* env)
+      : BaseObject(env, env->coverage_connection()),
+        env_(env),
+        session_(nullptr),
+        ending_(false) {
+    inspector::Agent* inspector = env->inspector_agent();
+    std::unique_ptr<inspector::InspectorSession> session =
+        inspector->Connect(std::unique_ptr<V8CoverageSessionDelegate>(
+                               new V8CoverageSessionDelegate(this)),
+                           false);
+    session_.reset(session.release());
```

Comment:
```suggestion
    session_ = std::move(session);
```

(assuming that works?)